src/containers/BooksList.js

```diff
@@ -2,35 +2,48 @@ import React from 'react';
 import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
 import Book from '../components/Book';
-import { removeCreator } from '../actions';
+import { removeCreator, filterCreator } from '../actions';
+import CategoryFilter from '../components/CategoryFilter';
 
-const mapStateToProps = state => ({ books: state.books });
+const filterHelper = event => event.target.value;
+
+const mapStateToProps = state => ({ books: state.books, filter: state.filter });
 
 const mapDispatchToProps = dispatch => ({
   handleRemoveBook: book => {
     dispatch(removeCreator(book));
   },
+  handleFilterChange: event => {
+    dispatch(filterCreator(filterHelper(event)));
+  },
 });
 
-const BooksList = ({ books, handleRemoveBook }) => (
-  <table>
-    { books.map(book => (
-      <tbody key={book.id}>
-        <Book book={book} />
-        <tr>
-          <td>
-            <button type="submit" onClick={() => handleRemoveBook(book)}> Remove Book </button>
-          </td>
-        </tr>
-      </tbody>
-    ))}
-  </table>
+const BooksList = ({
+  books, filter, handleRemoveBook, handleFilterChange,
+}) => (
+  <div>
+    <div>
+      <CategoryFilter handleFilterChange={handleFilterChange} />
+    </div>
+    <table>
+      {
+        filter === 'All'
+          ? books.map(book => (
+            <Book key={book.id} book={book} handleRemoveBook={handleRemoveBook} />
+          ))
+          : books.filter(book => (book.category === filter)).map(book => (
+            <Book key={book.id} book={book} handleRemoveBook={handleRemoveBook} />
```

Comment:
It is good to extract as much logic you can from the DOM, this check could have been done in a function above and the value called here :slightly_smiling_face:

Reply:
We were unsure about how to handle this part, we definitely apply this, thank you!!